### ISSUE TITLE
Transient Garbage Collection

### ISSUE THREAD

**Issue by jasonpierson** [oidc-wp/openid-connect-generic#279]
In openid-connect-generic.php, I see this function for garbage collection:
```
function cron_states_garbage_collection() {
	global $wpdb;
	$states = $wpdb->get_col( "SELECT `option_name` FROM {$wpdb->options} WHERE `option_name` LIKE '_transient_openid-connect-generic-state--%'" );

	if ( ! empty( $states ) ) {
		foreach ( $states as $state ) {
			$transient = str_replace( '_transient_', '', $state );
			get_transient( $transient );
		}
	}
}
```
Is the last line:
```
get_transient( $transient );
```
correct? Shouldn't it be:
```
set_transient( $transient );
```
?

**Comment by daggerhart** [comment]
Hi @jasonpierson 

This is correct as `get_transient()`. When gen_transient() runs and finds that a transient has expired, it will remove the transient automatically. We're allowing that function to decide whether or not the transient has expired, rather than doing it ourselves.

Are you running into some issue with transients not expiring?

**Comment by jasonpierson** [comment]
Thanks for the explanation @daggerhart ! I'm trying to troubleshoot a problem with session timeouts. Initially, we were getting an "invalid state" error after a few minutes of idle time; which was corrected with a higher timeout value. I set this at 36000 seconds b/c our IDP(Keycloak) expires login sessions after ten hours. This helped. I would say its now up to about four hours, but its still not at ten.
